### PR TITLE
CHANGELOG: add changes between 0.7.2 to current master

### PR DESCRIPTION
**Pull request by fanminshi**
Add major changes to the CHANGELOG from 0.7.2 tag to current master.

[skip ci]

### CODE REVIEW

**Comment by fanminshi** [comment]
cc/ @hasbro17 @hongchaodeng

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,11 +3,29 @@
 ### Added
 
 - Add readiness probe to etcd pod. The readiness state will be reflected on `status.members.ready/unready`.
+- Tls support in backup/restore-operator.
```
Comment (hongchaodeng) [inline CHANGELOG.md]: TLS etcd cluster ...

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,11 +3,29 @@
...
+- Add spec validation in restore operator.
+- Add BackupStorageType and EtcdClusterRef.
```
Comment (hongchaodeng) [inline CHANGELOG.md]: Add BackupStorageType to `EtcdRestore.Spec`. (Briefly what it does)
Add EtcdClusterRef ...

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,11 +3,29 @@
...
+- Add create-crd flag to etcd backup operator.
```
Comment (hongchaodeng) [inline CHANGELOG.md]: (Brief what it does)

**Comment by fanminshi** [comment]
all fixed. PTAL

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,11 +3,30 @@
 ### Added
 
 - Add readiness probe to etcd pod. The readiness state will be reflected on `status.members.ready/unready`.
+- TLS etcd cluster support in backup/restore-operator.
+- Add spec validation in restore operator.
+- Add BackupStorageType to EtcdRestore.RestoreSpec to indicate type of the backup storage which is used as RestoreSource.
+- Add EtcdClusterRef to EtcdRestore.RestoreSpec to reference an EtcdCluster resource which should have "Failed" status.phase and to be restored.
```
Comment (hongchaodeng) [inline CHANGELOG.md]: ~~which should have "Failed" status.phase and to be restored.~~ whose metadata and spec will be used to create the new restored EtcdCluster CR.

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,11 +3,30 @@
...
+- Use default etcd version when creating NewEtcdRestore.
+- Use S3 path instead of internal scheme
+- Randomize s3Path in e2e/backup_test.go
+- etcdbackup spec uses etcd endpoints to retrieve snapshot.
+- Change getEndpointWithMaxRev to getClientWithMaxRev.
+- Rename BackupCRStatus to BackupStatus.
+- Better error output in getClientWithMaxRev.
 - Default base image is changed to `gcr.io/etcd-development/etcd`, default etcd version is `3.2.13`.
 
+
```
Comment (hongchaodeng) [inline CHANGELOG.md]: delete blank line

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,11 +3,30 @@
 ### Added
 
 - Add readiness probe to etcd pod. The readiness state will be reflected on `status.members.ready/unready`.
+- TLS etcd cluster support in backup/restore-operator.
+- Add spec validation in restore operator.
+- Add BackupStorageType to EtcdRestore.RestoreSpec to indicate type of the backup storage which is used as RestoreSource.
+- Add EtcdClusterRef to EtcdRestore.RestoreSpec to reference an EtcdCluster resource which should have "Failed" status.phase and to be restored.
+- Add create-crd flag to etcd backup operator allowing user to disable automatic backup crd creation.
+- Add create-crd flag to etcd restore operator allowing user to disable automatic restore crd creation.
+- Add EtcdVersion and EtcdRevision to BackupStatus.
```
Comment (hongchaodeng) [inline CHANGELOG.md]: BackupStatus -> EtcdBackup.BackupStatus

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,11 +3,30 @@
...
+- Add validation of BackupStorageType in restore operator.
```
Comment (hongchaodeng) [inline CHANGELOG.md]: Merge this with `Add BackupStorageType` line?

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,11 +3,30 @@
...
+- Add run e2e tests from container.
```
Comment (hongchaodeng) [inline CHANGELOG.md]: Delete this line and the blank line

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,11 +3,30 @@
...
+
 
 ### Changed
 
+- Provide scripts to run e2e tests inside a pod.
+- Use default etcd version when creating NewEtcdRestore.
```
Comment (hongchaodeng) [inline CHANGELOG.md]: Delete 
`Provide scripts to run e2e tests inside a pod.` and 
`Use default etcd version when creating NewEtcdRestore.`

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,11 +3,30 @@
...
+- Use S3 path instead of internal scheme
```
Comment (hongchaodeng) [inline CHANGELOG.md]: EtcdBackup: BackupSpec passes in S3BackupSource.Path as the S3 path to save the backup. Get rid of previous internal scheme and remove S3 path in BackupStatus.

File: CHANGELOG.md [hunk 11]
```diff
@@ -3,11 +3,30 @@
...
+- Randomize s3Path in e2e/backup_test.go
```
Comment (hongchaodeng) [inline CHANGELOG.md]: Delete this

File: CHANGELOG.md [hunk 12]
```diff
@@ -3,11 +3,30 @@
...
+- etcdbackup spec uses etcd endpoints to retrieve snapshot.
```
Comment (hongchaodeng) [inline CHANGELOG.md]: EtcdBackup: BackupSpec ...

File: CHANGELOG.md [hunk 13]
```diff
@@ -3,11 +3,30 @@
...
+- Change getEndpointWithMaxRev to getClientWithMaxRev.
```
Comment (hongchaodeng) [inline CHANGELOG.md]: Delete this

File: CHANGELOG.md [hunk 14]
```diff
@@ -3,11 +3,30 @@
...
+- Rename BackupCRStatus to BackupStatus.
```
Comment (hongchaodeng) [inline CHANGELOG.md]: Move this before the S3 path change.

File: CHANGELOG.md [hunk 15]
```diff
@@ -3,11 +3,30 @@
...
+- Better error output in getClientWithMaxRev.
```
Comment (hongchaodeng) [inline CHANGELOG.md]: BackupStatus: Better error message for creating etcd client error.

**Comment by fanminshi** [comment]
all fixed. PTAL

**Comment by fanminshi** [comment]
PTAL cc/ @hongchaodeng

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**